cs169/BJC-Teacher-Tracker-App
# 187384622/new email model - PR3 - Add schema migration to drop old email columns and Adjust Relevant Dependencies

### [Pivotal Tracker Link][tracker]


[tracker]: https://www.pivotaltracker.com/story/show/187384622

## What this PR does:


This pull request is the **third** in a series intended to facilitate easier code review. The first PR can be found 
 [here](https://github.com/cs169/BJC-Teacher-Tracker-App/pull/49). The second PR can be found [here](https://github.com/cs169/BJC-Teacher-Tracker-App/pull/50).

For this PR, it aims to have a schema migration to drop the old email & personal_email column from `Teacher` model. It also adjust some old code that has dependencies on the old email fields. For example, `personal_email || email_addresses.where(primary: false)&.pluck(:email)&.first` now becomes `email_addresses.find_by(primary: true)&.email`

### Include screenshots, videos, etc.
#### Original DB State
<img width="1521" alt="image" src="https://github.com/cs169/BJC-Teacher-Tracker-App/assets/92573736/48f6ea18-bd74-465b-ad30-978105bd7e84">

#### DB State : After `rails db:migrate`

<img width="1502" alt="image" src="https://github.com/cs169/BJC-Teacher-Tracker-App/assets/92573736/fb435aa1-eadb-4d11-afda-f6eaae415291">

##### Frontend works fine as before
<img width="1547" alt="image" src="https://github.com/cs169/BJC-Teacher-Tracker-App/assets/92573736/a98bc875-ef56-4588-bedc-da44ead64025">

#### Who authored this PR?

1. Perry (Jingchao) Zhong @perryzjc 


### How should this PR be tested?
1. Test the app work fine as before after dropping the old email columns
2. Also would be nice to ensure the code is clean that has no dependencies on the old email column anymore.

* Is there a deploy we can view?
* What do the specs/features test?
* Are there edge cases to watch out for?

#### Are there any complications to deploying this?
Yes. Run `rails db:migrate` to apply the migration.



### Checklist:
